File: models/token.go
package models

import (
	"math/big"

	"github.com/ethereum/go-ethereum/common"
)

// Token represents the model of a token
type Token struct {
	Name     string
	Symbol   string
	Decimals uint64
	Address  *common.Address
}

// TokenAddressWithBalance represents the model of a token address with its balance
type TokenAddressWithBalance struct {
	Address *common.Address
	Balance *big.Int
}

File: src/models/token.ts
/**
 * Token model
 */
export class Token {
  id: string = "";
  name: string = "";
  decimals: number = 0;
  symbol: string = "";
  address: string = "";

  constructor(token: Token) {
    this.id = token.id || "";
    this.name = token.name || "";
    this.symbol = token.symbol || "";
    this.address = token.address || "";
    this.decimals = token.decimals || 0;
  }
}
